### PR TITLE
update to ament_cmake and added env-hooks

### PR DESCRIPTION
**Pull request by joe28965**
Issue #3

The foxy-devel branch still uses the old ROS1 structure. I have updated the cmakefile, package.xml and launch files to reflect the new updated ROS2 structure.

I have also added [env-hooks](https://github.com/ros-simulation/gazebo_ros_pkgs/wiki/ROS-2-Migration:-Gazebo-ROS-Paths#env-hooks). These will automatically set your GAZEBO_MODEL_PATH when you source your workspace. This will eliminate the need to use:

`export GAZEBO_MODEL_PATH=`pwd`/models`

every time you open a terminal (or the need to add it to your .bashrc)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mm318** [comment]
It looks like this pull request will conflict with https://github.com/aws-robotics/aws-robomaker-small-warehouse-world/pull/5. Let's reconcile that, and make adding the env-hooks a separate pull request.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+################################################################################
+# Install
+################################################################################
+install(
+  DIRECTORY launch models worlds maps rviz
+  DESTINATION share/${PROJECT_NAME}
 )
 
+ament_environment_hooks("${CMAKE_CURRENT_SOURCE_DIR}/env-hooks/aws_robomaker_small_warehouse_world.dsv.in")
+
```
Comment (ojasjoshi) [inline CMakeLists.txt:22]: Add export dependencies for downstream packages using this world
```
ament_export_dependencies(ament_cmake)
ament_export_dependencies(gazebo_ros)
```

Comment (joe28965) [reply CMakeLists.txt:22]: I have added them

**Comment by joe28965** [comment]
I also saw the other pull request. However that uses the gazebo_ros launch files for gzserver and client in order to start gazebo. I personally prefer it the way I did (although I am biased of course, since I chose to do it this way), because updates in gazebo_ros will not cause problems with this repo.

**Comment by padiln** [comment]
@joe28965 @mm318 
It have no hard feelings closing my PR and just pull #4 pending approval.  No need to worry about merging since they do about the same thing.

**Comment by joe28965** [comment]
These changes have been added to PR #5 making this PR obsolete. I will close this to avoid confusion.

**Pull request closed without merging**